#550 cmake build is missing key features to be properly usable via fetchcontent

globberwops opened the pull request:
﻿#### Reference to a related issue in the repository
Resolves #550 

#### Add a description
This PR aims to modernize the installation logic of the project by:
* removing redundant warnings
* prefix cache variables with the project name
* provide namespaced alias targets usable from the build tree
* provide a namespace for the export set

This modernization will make the project usable via [FetchContent](https://cmake.org/cmake/help/latest/module/FetchContent.html) by a downstream project.

#### Take this checklist as orientation for yourself, if this PR is ready for the Change Control Board:
- [x] My suggestion follows the [style and contributors guidelines](https://opensimulationinterface.github.io/osi-documentation/open-simulation-interface/doc/howtocontribute.html).
- [x] I have taken care about the [documentation](https://opensimulationinterface.github.io/osi-documentation/open-simulation-interface/doc/commenting.html).
- [x] I have done the [DCO signoff](https://opensimulationinterface.github.io/osi-documentation/open-simulation-interface/doc/howtocontribute.html#developer-certification-of-origin-dco).
- [x] My changes generate no errors when passing CI tests. 
- [x] I have successfully implemented and tested my fix/feature locally.
- [x] Appropriate reviewer(s) are assigned.

tbleher commented:
Thank you for the CMake improvements! I have not used all the CMake constructs you use here, but from what I see this all looks sensible :)

ThomasNaderBMW reviewed (approved):
Thank you @tbleher for the feedback.

From my side: I tested the basic functionality, so to build the osi-project and had no issues.

So together a +1 :)

globberwops commented:
@tbleher and @ThomasNaderBMW Thanks for your feedback.
I have merged the latest master branch and fixed the new flatbuffers option.
C++11 is now required, please tell me if this is intended or not.
I have also added the toplevel project check to restrict the doc target.

LukasElster commented:
Our simulation models are still building with the changes. Therefore the changes are fine for me

kmeids commented:
**Output CCB 19.01.2022:**
1. @pmai  to check the effect/dependencies of these changes on OSMP/OSI-Validator.

kmeids commented:
**Output CCB 02.02.2022:**
1. @pmai  to check the effect/dependencies of these changes on OSMP/OSI-Validator.

kmeids commented:
**Output CCB 25.04.2022:** 
1. @pmai and @ThomasNaderBMW, could you please bring this request to the SetLevel agenda to check if it matches with the latest changes?.

ThomasNaderBMW commented:
Output CCB 16.05.2022: Can be merged.
